lib/ansible/plugins/test/core.py

```diff
@@ -170,6 +171,8 @@ def version_compare(value, version, operator='eq', strict=False):
     try:
         method = getattr(py_operator, operator)
         return method(Version(str(value)), Version(str(version)))
+    except UndefinedError as e:
+        raise errors.AnsibleUndefinedVariable('Version comparison: %s' % e)
```

Comment:
At first I thought we could just re-raise the `UndefinedError` and let `do_template()` deal with that but converting to our exception early makes more sense. :+1:

Reply:
Yeah, I waffled back and forth about that too.

Reply:
I ended up changing this after all - with my latest change our filters are no longer intercepting the exception, just triggering the right one earlier